src/main/java/org/opensearch/neuralsearch/processor/DocumentChunkingProcessor.java

```diff
@@ -0,0 +1,224 @@
+                        + algorithmKey
+                        + "] is not supported. Supported chunkers types are ["
+                        + FIXED_LENGTH_ALGORITHM
+                        + ", "
+                        + DELIMITER_ALGORITHM
+                        + "]"
+                );
+            }
+            if (!(algorithmValue instanceof Map)) {
+                throw new IllegalArgumentException(
+                    "Unable to create the processor as [" + ALGORITHM_FIELD + "] cannot be cast to [" + Map.class.getName() + "]"
+                );
+            }
+            this.chunkerType = algorithmKey;
+            this.chunkerParameters = (Map<String, Object>) algorithmValue;
+        }
+    }
+
+    @Override
+    protected List<?> buildResultForListType(List<Object> sourceValue, List<?> results, IndexWrapper indexWrapper) {
+        Object peek = sourceValue.get(0);
+        if (peek instanceof String) {
+            List<Object> keyToResult = new ArrayList<>();
+            IntStream.range(0, sourceValue.size()).forEachOrdered(x -> keyToResult.add(results.get(indexWrapper.index++)));
+            return keyToResult;
+        } else {
+            List<List<Object>> keyToResult = new ArrayList<>();
+            for (Object nestedList : sourceValue) {
+                List<Object> nestedResult = new ArrayList<>();
+                IntStream.range(0, ((List) nestedList).size()).forEachOrdered(x -> nestedResult.add(results.get(indexWrapper.index++)));
+                keyToResult.add(nestedResult);
+            }
+            return keyToResult;
+        }
+    }
+
+    @Override
+    public void doExecute(
+        IngestDocument ingestDocument,
+        Map<String, Object> ProcessMap,
+        List<String> inferenceList,
+        BiConsumer<IngestDocument, Exception> handler
+    ) {
+        try {
+            processorInputValidator.validateFieldsValue(fieldMap, environment, ingestDocument, true);
+            if (Objects.equals(chunkerType, FIXED_LENGTH_ALGORITHM)) {
+                // add maxTokenCount setting from index metadata to chunker parameters
+                Map<String, Object> sourceAndMetadataMap = ingestDocument.getSourceAndMetadata();
+                String indexName = sourceAndMetadataMap.get(IndexFieldMapper.NAME).toString();
+                int maxTokenCount = IndexSettings.MAX_TOKEN_COUNT_SETTING.get(settings);
+                IndexMetadata indexMetadata = clusterService.state().metadata().index(indexName);
+                if (indexMetadata != null) {
+                    // if the index exists, read maxTokenCount from the index setting
+                    IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
+                    maxTokenCount = indexService.getIndexSettings().getMaxTokenCount();
+                }
+                chunkerParameters.put(FixedTokenLengthChunker.MAX_TOKEN_COUNT_FIELD, maxTokenCount);
```

Comment:
Will this influence the next ingest? If all documents belong to same index, the logic is ok but we can only acquire the maxTokenCount once. Otherwise, we need to remove this field after one processing.